### PR TITLE
Remove Python 2 code from C extensions

### PR DESCRIPTION
**Pull request by QuLogic**
## PR Summary

Remove a bunch of fallback code and macros that provided Python 2 compatibility in the C extensions. ~~Note, this is based on #7853, #10426 and #10383 because I did not want to work out the conflicts later.~~ I'm putting this out there to prevent duplicate work later.

## PR Checklist

- [N/A] Has Pytest style unit tests
- [x] Code is PEP 8 compliant
- [N/A] New features are documented, with examples if plot related
- [N/A] Documentation is sphinx and numpydoc compliant
- [N/A] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [N/A] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by QuLogic** [comment]
I'm going to mark this WIP not because it is, but just so that it doesn't accidentally get merged before the other 3.

File: src/mplutils.h [hunk 1]
```diff
@@ -31,10 +31,7 @@ typedef unsigned __int8   uint8_t;
 #include <Python.h>
 
 #if PY_MAJOR_VERSION >= 3
-#define PY3K 1
```
Comment (anntzer) [inline src/mplutils.h]: Py_TPFLAGS_HAVE_NEWBUFFER can also be removed throughout the codebase

File: src/_tkagg.cpp [hunk 2]
```diff
@@ -326,7 +326,6 @@ int load_tkinter_funcs(void)
  * tkinter uses these symbols, and the symbols are therefore visible in the
  * tkinter dynamic library (module).
  */
-#if PY_MAJOR_VERSION >= 3
```
Comment (anntzer) [inline src/_tkagg.cpp]: just inline the conditional stuff below?

File: src/mplutils.h [hunk 3]
```diff
@@ -30,9 +30,7 @@ typedef unsigned __int8   uint8_t;
 
 #include <Python.h>
 
-#if PY_MAJOR_VERSION >= 3
```
Comment (anntzer) [inline src/mplutils.h]: inline the macro

File: src/_tkagg.cpp (hunk 2 above)
Comment (QuLogic) [reply src/_tkagg.cpp]: OK, will do when I next rebase.

**Comment by efiring** [comment]
This is now waiting only on #10426, which looks ready to go after a simple merge conflict fixup.

**Comment by QuLogic** [comment]
Rebased now that everything else is merged.

**Pull request merged**